### PR TITLE
chore: add snapshots for input type widgets

### PR DESCRIPTION
**Pull request by jsartisan**
/ok-to-test tags="@tag.Anvil"

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/9678081244>
> Commit: d73f2d7f37c9d62ba1ca622b78d9a0ce4d0e3db8
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=9678081244&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Anvil`

## Summary by CodeRabbit

- **New Features**
  - Introduced comprehensive test cases for Currency Input, Phone Input, and General Input Widgets in the Anvil editor covering Canvas, Preview, and Deploy modes.
  - Added complex layout structures for Currency and Phone Input Widgets in the "MainContainer" canvas widget.

- **Enhancements**
  - Implemented debouncing for validation status and error message handling in the Currency Input Widget to improve timing accuracy.
  - Added `defaultValue` property to Currency Input Widget configurations.
  - Corrected typos and improved clarity in widget property names and validation logic.

- **Bug Fixes**
  - Fixed validation status typo in Phone Input Widget.
  
- **Refactor**
  - Reordered static property declarations in Input Widgets for better code organization.
  - Modified method access controls in AnvilSnapshot for external usage.
  
- **Tests**
  - Added snapshot testing for Currency, Phone, and General Input Widgets within the Anvil editor.
  
- **Chores**
  - Added a new method `Createpage(pageName: string)` to streamline page creation in tests.

### CODE REVIEW

File: app/client/cypress/snapshots/AnvilCurrencyInputWidgetSnapshot_spec.ts/anvilCurrencyInputWidgetCanvas.snap.png
Comment (KelvinOm) [inline app/client/cypress/snapshots/AnvilCurrencyInputWidgetSnapshot_spec.ts/anvilCurrencyInputWidgetCanvas.snap.png:1]: 1. Can we check invalid state here as well?
2. Can we add inputs into 1 2 3 4 zones?
3. Can we add really long value?

Comment (jsartisan) [inline app/client/cypress/snapshots/AnvilCurrencyInputWidgetSnapshot_spec.ts/anvilCurrencyInputWidgetCanvas.snap.png:1]: Let me fix this snapshot.

File: app/client/src/widgets/wds/WDSPhoneInputWidget/widget/helpers.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ export function validateInput(props: any) {
   }
 
   return {
-    validattionStatus: isInvalid ? "invalid" : undefined,
```
Comment (jsartisan) [inline app/client/src/widgets/wds/WDSPhoneInputWidget/widget/helpers.ts:21]: Spelling was wrong.

File: app/client/src/widgets/wds/WDSPhoneInputWidget/widget/index.tsx [hunk 2]
```diff
@@ -17,6 +16,7 @@ import { WDSBaseInputWidget } from "widgets/wds/WDSBaseInputWidget";
 import { AsYouType, parseIncompletePhoneNumber } from "libphonenumber-js";
 import { EventType } from "constants/AppsmithActionConstants/ActionConstants";
 import type { KeyDownEvent } from "widgets/wds/WDSBaseInputWidget/component/types";
+import derivedProperties from "widgets/wds/WDSCurrencyInputWidget/widget/parsedDerivedProperties";
```
Comment (jsartisan) [inline app/client/src/widgets/wds/WDSPhoneInputWidget/widget/index.tsx:19]: reusing the dervied logic of currency widget.

File: app/client/src/widgets/wds/WDSPhoneInputWidget/widget/index.tsx [hunk 3]
```diff
@@ -310,7 +310,7 @@ class WDSPhoneInputWidget extends WDSBaseInputWidget<
         onValueChange={this.onValueChange}
         placeholder={this.props.placeholderText}
         tooltip={this.props.tooltip}
-        validationStatus={validation.validattionStatus}
```
Comment (jsartisan) [inline app/client/src/widgets/wds/WDSPhoneInputWidget/widget/index.tsx:313]: fixing the wrong spelling here.

**Review by KelvinOm — APPROVED** [review]
LGTM.

@jsartisan Should we ask @ramsaptami to check your changes in inputs?

**Comment by jsartisan** [comment]
> LGTM.
> 
> @jsartisan Should we ask @ramsaptami to check your changes in inputs?

I think it's fine. I just updated the validation.

**Review by KelvinOm — APPROVED** [review]

**Pull request merged**